pkg/kube/apisix_tls.go

```diff
@@ -0,0 +1,167 @@
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package kube
+
+import (
+	"fmt"
+
+	"github.com/apache/apisix-ingress-controller/pkg/config"
+	configv2 "github.com/apache/apisix-ingress-controller/pkg/kube/apisix/apis/config/v2"
+	configv2beta3 "github.com/apache/apisix-ingress-controller/pkg/kube/apisix/apis/config/v2beta3"
+	listersv2 "github.com/apache/apisix-ingress-controller/pkg/kube/apisix/client/listers/config/v2"
+	listersv2beta3 "github.com/apache/apisix-ingress-controller/pkg/kube/apisix/client/listers/config/v2beta3"
+)
+
+// ApisixTlsLister is an encapsulation for the lister of ApisixTls,
+// it aims at to be compatible with different ApisixTls versions.
+type ApisixTlsLister interface {
+	// V2beta3 gets the ApisixTls in apisix.apache.org/v2beta3.
+	V2beta3(string, string) (ApisixTls, error)
+	// V2 gets the ApisixTls in apisix.apache.org/v2.
+	V2(string, string) (ApisixTls, error)
+}
+
+// ApisixTlsInformer is an encapsulation for the informer of ApisixTls,
+// it aims at to be compatible with different ApisixTls versions.
+type ApisixTlsInformer interface {
+	Run(chan struct{})
+}
+
+// ApisixTls is an encapsulation for ApisixTls resource with different
+// versions, for now, they are apisix.apache.org/v1 and apisix.apache.org/v2alpha1
+type ApisixTls interface {
+	// GroupVersion returns the api group version of the
+	// real ApisixTls.
+	GroupVersion() string
+	// V2beta3 returns the ApisixTls in apisix.apache.org/v2beta3, the real
+	// ApisixTls must be in this group version, otherwise will panic.
+	V2beta3() *configv2beta3.ApisixTls
+	// V2 returns the ApisixTls in apisix.apache.org/v2, the real
+	// ApisixTls must be in this group version, otherwise will panic.
+	V2() *configv2.ApisixTls
+	// ResourceVersion returns the the resource version field inside
+	// the real ApisixTls.
+	ResourceVersion() string
+}
+
+// ApisixTlsEvent contains the ApisixTls key (namespace/name)
+// and the group version message.
+type ApisixTlsEvent struct {
+	Key          string
+	OldObject    ApisixTls
+	GroupVersion string
+}
+
+type apisixTls struct {
+	groupVersion string
+	v2beta3      *configv2beta3.ApisixTls
+	v2           *configv2.ApisixTls
+}
+
+func (ar *apisixTls) V2beta3() *configv2beta3.ApisixTls {
```

Comment:
If the meaning of `ar` is `APISIX resource` , then we can go ahead and use it. 
Otherwise a clearer name should be used, right?